cmake/modules/autopas_clang-sanitizers.cmake

```diff
@@ -1,6 +1,18 @@
-option(AUTOPAS_ENABLE_ADDRESS_SANITIZER "Adds clang's address sanitizer to the build options" OFF)
-option(AUTOPAS_ENABLE_MEMORY_SANITIZER "Adds clang's memory sanitizer to the build options" OFF)
-option(AUTOPAS_ENABLE_THREAD_SANITIZER "Adds clang's thread sanitizer to the build options" OFF)
+cmake_dependent_option(AUTOPAS_ENABLE_ADDRESS_SANITIZER "Adds clang's address sanitizer to the build options"
+        OFF
+        AUTOPAS_STANDALONE_BUILD
+        OFF
+)
+cmake_dependent_option(AUTOPAS_ENABLE_MEMORY_SANITIZER "Adds clang's memory sanitizer to the build options"
+        OFF
+        AUTOPAS_STANDALONE_BUILD
+        OFF
+)
+cmake_dependent_option(AUTOPAS_ENABLE_THREAD_SANITIZER "Adds clang's thread sanitizer to the build options"
+        OFF
+        AUTOPAS_STANDALONE_BUILD
+        OFF
+)
```

Comment:
These can actually make sense when used in other programs.
mardyn, e.g., sets those.
if they cannot be set from outside, potential internal errors of autopas' compiled files cannot be found anymore.

Reply:
They can be set from the outside, they are just not visible in any cmake gui until you set `AUTOPAS_STANDALONE_BUILD=ON` which you can also do explicitly. But we maybe should document this somewhere.

Reply:
See custom implementation in 2b782bc and its application in e0b2429